usage/usageplot.py

```diff
@@ -85,23 +93,107 @@ def make_plot(title, times, nodes, name):
+                                 nodes[val_tup[0]:val_tup[1]+2], hatch=hatch[key],
+                                 step="post", facecolor=colors[key], alpha=0.5,
+                                 label=key if key not in
+                                 plt.gca().get_legend_handles_labels()[1] else '')
+        plt.legend(title='Code Name')
+    
+    else:
+        plt.fill_between(times, 0, nodes, step="post", facecolor='b', alpha=0.25)
+
     plt.xlabel(r'time [h]', fontsize=16)
     plt.ylabel(r'$\ N_{node} $', fontsize=16)
     plt.tick_params(axis='both', which='major', labelsize=14)
     plt.title(title, fontsize=18)
     plt.savefig(name + ".png")
 
 
+def get_first_last(jobs):
+    """Gets the first and the last index of the code names of the same type,
+    (i.e.: will return the first and last index of the singleFrameDriver jobs)
+    which will allow for color-coding in make-plot.
+
+    Parameters
+    ----------
+    jobs : `list` of `str`
+        list of the job names from SLURM at a given time step
+
+     Returns
+    -------
+    start_end : `dict` of `list` of `tuple`
+        dictionary containing code names and the start and stop indexes
+        of the code runs.
+    """
+    names = dict()
+    # Creates dictionary of the first two letters of a job name & the indexes
+    # where those job names can be found.
+    for k in range(len(jobs)):
+        for j in jobs[k]:
+            if j[:2] in names:
+                names[j[:2]].append(k)
+            else:
+                names[j[:2]] = [k]
+
+    data_final = dict()
+    # Translates common entries in the names dictionary to the actual code
+    # names. If the key in names doesn't match with any of those below, the
+    # code will be marked as "unknown"
+    for key in names.keys():
+        if key == 'Wi':
+            data_final['singleFrame'] = names[key]
+        elif key == 'Co':
+            data_final['singleFrame'] += names[key]
+        elif 'mo' == key:
+            data_final['mosaic'] = names[key]
+        elif 'co' == key:
+            data_final['coadd'] = names[key]
+        elif 'mt' == key:
+            data_final['multiband'] = names[key]
+        else:
+            data_final['unknown'] = names[key]
+
+    # Removes repeated index values for each key and sorts the values into
+    # ascending order.
+    data_final = {k: sorted(list(set(v))) for (k, v) in data_final.items()}
+
+    start_end = dict()
+    # Creates a dictionary only listing tuples of the first and last indexes of
+    # when a code was run.  If there are gaps in the indexes, then there will
+    # be multiple tuples for each key, thus denoting when that code started and
+    # stopped. This is done to make the shading in make_plot more accurate.
+    for key, value in data_final.items():
+        gap = [[s, e] for s, e, in zip(value, value[1:]) if s + 1 < e]
+        edges = iter(value[:1] + sum(gap, []) + value[-1:])
+        start_end[key] = list(zip(edges, edges))
+
```

Comment:
Nice trick with grouping list elements pairwise. I didn't know it. I'd probably would go for something more obvious
```python
edges = value[:1] + sum(gap, []) + value[-1:]
start_end[key] = list(zip(edges[::2], edges[1::2]
```